src/web/jcc/client/components/SetHubLocationPanel.tsx

```diff
@@ -0,0 +1,188 @@
+        const lat = Number(getElementById<HTMLInputElement>('set-hub-location-latitude').value)
+        const lon = Number(getElementById<HTMLInputElement>('set-hub-location-longitude').value)
+        if (lat == null || lon == null) {
+            return null
+        }
+
+        return {
+            lat: lat,
+            lon: lon
+        }
+    }
+
+
+    /**
+     * Calls the API to submit a location change for a certain hub.
+     *
+     */
+    function submitHubLocation() {
+        const hubLocation = getInputHubLocation()
+
+        if (hubLocation == null) {
+            console.warn('hub location is null')
+            return
+        }
+
+        const hubCommand: CommandForHub = {
+            hub_id: hub_id,
+            type: HubCommandType.SET_HUB_LOCATION,
+            set_hub_location: hubLocation
+        }
+
+        props.api.postCommandForHub(hubCommand)
+    }
+
+    
+    /**
+     * Initiate a PointerInteraction to select a new hub location with a click or tap.
+     */
+    const toggleSelectOnMapInteraction = () => {
+        if (selectOnMapInteractionRef.current != null) {
+            destroySelectOnMapInteraction()
+            return
+        }
+
+        selectOnMapInteractionRef.current = new PointerInteraction({
+            handleEvent: (evt) => {
+                if (evt.type == "click") {
+                    const clickedLocation = getGeographicCoordinate(evt.coordinate, evt.map)
+                    getElementById<HTMLInputElement>('set-hub-location-latitude').value = clickedLocation.lat.toFixed(6)
+                    getElementById<HTMLInputElement>('set-hub-location-longitude').value = clickedLocation.lon.toFixed(6)
+                    submitHubLocation()
+                    destroySelectOnMapInteraction()
+                    return false
+                }
+            },
+            stopDown: () => { return true }
+        })
+        setSelectingOnMap(true)
+        props.map.addInteraction(selectOnMapInteractionRef.current)
+        props.map.getTargetElement().style.cursor = 'crosshair'
+    }
+    
+
+    /**
+     * Destroy the hub location selection interaction.
+     */
+    const destroySelectOnMapInteraction = () => {
+        if (selectOnMapInteractionRef.current != null) {
+            props.map.removeInteraction(selectOnMapInteractionRef.current)
+            selectOnMapInteractionRef.current = null
+            props.map.getTargetElement().style.cursor = 'default'
+            setSelectingOnMap(false)
+        }
+    }
+
+
+    // Destroy the map select interaction on unmount, if present.
+    useEffect(() => {
+        return destroySelectOnMapInteraction
+    }, [])
+
+
+    return (
+        <div className="panel" key="set-hub-location">
+            <div className='panel-heading' style={{color: "black"}}>Set Hub Location</div>
+            <div className='mission-settings-panel-container'>
+
+                { /* this.hubIdSelectionElement() */ }
```

Comment:
Is this meant to be commented out? If so, it doesn't look like we need the hubIdSelectionElement() function

Reply:
It is commented out, because as @michael-jaia noted, the sim doesn't allow multi-hub yet.  I kept the function around in case we want it someday.